tests/integration/test_rex.py

```diff
@@ -708,7 +708,7 @@ def test_date_functions(c):
             "plus_1_sec": [datetime(2021, 10, 3, 15, 53, 43, 47)],
             "plus_999_millisec": [datetime(2021, 10, 3, 15, 53, 42, 1000 * 999 + 47)],
             "plus_999_microsec": [datetime(2021, 10, 3, 15, 53, 42, 1046)],
-            "plus_1_qt": [datetime(2022, 1, 3, 15, 53, 42, 47)],
+            "plus_1_qt": [datetime(2022, 1, 2, 15, 53, 42, 47)],
```

Comment:
The only thing about this solution is that because we aren't iterating through the rows, we can't check that the days stay consistent, so there's a rounding error. I tried seeing if we could use some sort of `replace` function with the days, but was having trouble getting it to work.

For example:
```
SELECT
    TIMESTAMPADD(MONTH, 1, d) as "plus_1_month",
    TIMESTAMPADD(MONTH, 2, d) as "plus_2_months",
    TIMESTAMPADD(MONTH, 6, d) as "plus_6_months",
    TIMESTAMPADD(MONTH, 36, d) as "plus_36_months",
    TIMESTAMPADD(QUARTER, 1, d) as "plus_1_qt",
    TIMESTAMPADD(QUARTER, 2, d) as "plus_2_qts",
    TIMESTAMPADD(QUARTER, 6, d) as "plus_6_qts",
    TIMESTAMPADD(QUARTER, 36, d) as "plus_36_qts"
FROM df
```

produces:
<html>
<body>
<!--StartFragment-->

  | plus_1_month | plus_2_months | plus_6_months | plus_36_months | plus_1_qt | plus_2_qts | plus_6_qts | plus_36_qts
-- | -- | -- | -- | -- | -- | -- | -- | --
0 | 2021-11-03 01:53:42.000047 | 2021-12-03 11:53:42.000047 | 2022-04-04 03:53:42.000047 | 2024-10-02 15:53:42.000047 | 2022-01-02 21:53:42.000047 | 2022-04-04 03:53:42.000047 | 2023-04-04 03:53:42.000047 | 2030-10-01 15:53:42.000047

<!--EndFragment-->
</body>
</html>

While the *actual* expected output would be:
<html>
<body>
<!--StartFragment-->

  | plus_1_month | plus_2_months | plus_6_months | plus_36_months | plus_1_qt | plus_2_qts | plus_6_qts | plus_36_qts
-- | -- | -- | -- | -- | -- | -- | -- | --
0 | 2021-11-03 15:53:42.000047 | 2021-12-03 15:53:42.000047 | 2022-04-03 15:53:42.000047 | 2024-10-03 15:53:42.000047 | 2022-01-03 15:53:42.000047 | 2022-04-03 15:53:42.000047 | 2023-04-03 15:53:42.000047 | 2030-10-03 15:53:42.000047

<!--EndFragment-->
</body>
</html>

So the question is how much do we care about rounding errors? I'd also be happy to chat about other possible solutions.